lib/index.js

```diff
@@ -170,7 +178,7 @@ internals.Client.prototype._request = function (method, url, options, relay, _tr
         uri.agent = options.agent;
     }
     else {
-        uri.agent = uri.protocol === 'https:' ? this.agents.https : this.agents.http;
+        uri.agent = parsedUri.protocol === 'https:' ? this.agents.https : this.agents.http;
```

Comment:
can this change to `uri`?

Reply:
Done. I finally did a `findAll` on parsedUri and I think the only 2 uses left are the needed ones.

Reply:
You rock, thanks for this PR... I'll release as 14.2.0. I'm going to bring in the changes from the other open PR as well around adding a new event.